### PR TITLE
 Issue #SB-421 feat: Announcement workflow

### PR DESCRIPTION
**Pull request by souravdey091**
1) Announcement Inbox/Outbox listing
2) Create announcement
3) Announcement details

### CODE REVIEW

File: src/app/private/scripts/filters/isValidObjectFilter.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+angular.module('playerApp')
```
[the remark this answers is not in the document]
Comment (niteshkesarkar) [reply src/app/private/scripts/filters/isValidObjectFilter.js]: Removed the filter. Used `ng-show` and `ng-hide` in the replacement of it.

File: src/app/private/scripts/controllers/announcement/createAnnouncementController.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        	createAnn.enableRecepientBtn();
+    	}
+
+    	// Function to detect dropdwon value change event
+    	createAnn.detectDropdownChange = function (){
+    		createAnn.enableRecepientBtn();
+    	}
+
+    	// Function to track back button change
+    	createAnn.previousStep = function(item){
```
[the remark this answers is not in the document]
Comment (nilesh-more) [reply src/app/private/scripts/controllers/announcement/createAnnouncementController.js]: @harishgilimi resolved

File: src/app/private/config/playerAppConfig.json [hunk 3]
```diff
@@ -96,7 +96,7 @@
                 "contentTypes": [
                     "Collection", "LessonPlan", "Resource", "TextBook"
                 ],
-                "resourceType": [ 
+                "resourceType": [
```
[the remark this answers is not in the document]
Comment (nilesh-more) [reply src/app/private/config/playerAppConfig.json]: @harishgilimi there was an empty space, and it is removed now

**Comment by harishgilimi** [comment]
@nilesh-more @souravdey091 This branch has conflicts that must be resolved

File: src/app/private/views/announcement/announcementInboxList.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<div class="ui grid" ng-init="announcementInboxData.renderAnnouncementList()">
+  <div class="one wide column"></div>
+  <div class="ten wide column">
+    <div class="ui grid">
+      <div class="eight wide column">
+        <div class="ui">
+          <div class="text dashBoardSectionHeading margin-bottom-20"><i class=" arrow left icon padding-right-10"></i> All Announcements</div>
```
Comment (loganathan1989) [inline src/app/private/views/announcement/announcementInboxList.html]: load message 'All Announcements' from resource bundles

File: src/app/private/views/announcement/announcementInboxListCard.html [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            </p>
+        </div>
+        <div class="padding-top-5">
+            <span class="ng-binding announcement-description">
+                {{item.details.description | limitTo:180}}
+            </span>
+            <span ng-show="item.description.length > 180">
+                ...
+                <span class="announcementBlueText">
+                    Read More
```
Comment (loganathan1989) [inline src/app/private/views/announcement/announcementInboxListCard.html]: user existing or add new entry in resource bundles for test 'Read More'

File: src/app/private/views/announcement/announcementOutboxList.html [hunk 6]
```diff
@@ -0,0 +1,98 @@
+<div ng-init="announcementOutboxData.renderAnnouncementList()">
+  <div class="ui grid">
+    <div class="one wide column"></div>
+    <div class="ten wide column">
+      <div class="ui grid">
+        <div class="nine wide column">
+          <div class="ui">
+            <div class="text dashBoardSectionHeading padding-top-10"><i class="announcement icon padding-right-10"></i> My Announcements</div>
```
Comment (loganathan1989) [inline src/app/private/views/announcement/announcementOutboxList.html]: user resource bundles for text 'My Announcements'

File: src/app/private/views/announcement/announcementOutboxList.html [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      </div>
+    </div>
+  </div>
+  <div class="ui grid" ng-if="announcementOutboxData.showDataDiv">
+    <div class="one wide column"></div>
+    <div class="ten wide column">
+      <div class="ui grid">
+        <div class="eight wide column">
+          <div class="ui">
+            <div class="text padding-top-5 showingAllSentAnn">Showing all sent announcements</div>
```
Comment (loganathan1989) [inline src/app/private/views/announcement/announcementOutboxList.html]: use resource bundles for text  'Showing all sent announcements'

File: src/app/private/views/announcement/announcementOutboxList.html [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    <div class="ten wide column">
+      <div class="ui grid">
+        <div class="nine wide column">
+          <div class="ui">
+            <div class="text dashBoardSectionHeading padding-top-10"><i class="announcement icon padding-right-10"></i> My Announcements</div>
+          </div>
+        </div>
+        <div class="right floated three wide column padding-right-dropdown padding-right-10">
+          <div class="ui float-ContentRight" ng-controller="createAnnouncementCtrl as createAnn">
+            <button class="ui primary basic button announcementButton" id="openmodal" ng-click="createAnn.createAnnouncement();"><span class="announcementButtonText">CREATE ANNOUNCEMENT</span></button>
```
Comment (loganathan1989) [inline src/app/private/views/announcement/announcementOutboxList.html]: load button text 'CREATE ANNOUNCEMENT' from resource bundles

File: src/app/private/views/announcement/announcementOutboxList.html [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    <div class="one wide column"></div>
+    <div class="ten wide column">
+      <app-loader></app-loader>
+    </div>
+  </div>
+  <div class="ui grid" ng-if="!announcementOutboxData.showLoader" ng-hide="announcementOutboxData.showDataDiv">
+    <div class="one wide column"></div>
+    <div class="ten wide column">
+      <div class="ui warning message">
+        <div class="header">No announcement found!</div>
```
Comment (loganathan1989) [inline src/app/private/views/announcement/announcementOutboxList.html]: load static text from resource bundles

File: src/app/private/scripts/controllers/announcement/announcementInboxListController.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+              announcementInboxData.result = apiResponse.result
+              announcementInboxData.listData = apiResponse.result.announcements
+              var page = false;
+              angular.forEach(announcementInboxData.listData, function(value, key) {
+                if (!page) {
+                  if (key + 1 === announcementInboxData.limit) {
+                    page = true;
+                  }
+                  // Call received API
+                  announcementService.receivedAnnouncement($rootScope.userId, value.id).then(function(apiResponse) {})
```
Comment (loganathan1989) [inline src/app/private/scripts/controllers/announcement/announcementInboxListController.js]: 'apiResponse' is not processed in success block.

**Pull request merged**